res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -281,22 +281,26 @@ DDJ200.cueGotoandstop = function(channel, control, value, status, group) {
 };
 
 DDJ200.hotcueNActivate = function(channel, control, value, status, group) {
-    var vDeckNo = DDJ200.vDeckNo[script.deckFromGroup(group)];
-    var vgroup = "[Channel" + vDeckNo + "]";
-    var hotcue = "hotcue_" + (control + 1);
-    engine.setValue(vgroup, hotcue + "_activate", true);
-    midi.sendShortMsg(status, control,
-        0x7F * engine.getValue(vgroup, hotcue + "_enabled"));
-    var deckNo = script.deckFromGroup(group);
-    midi.sendShortMsg(0x90 + deckNo - 1, 0x0B, 0x7F *
-                      engine.getValue(vgroup, "play")); // set play LED
+    if (value) { // only if button pressed, not releases, i.e. value === 0
```

Comment:
I think we should early exit here:


```suggestion
    if (!value) {
        return;
    }

```

Reply:
Generally this would be an option, but in the rest of the code I was doing it this way.
So it could be a bit inconsistent and confusing if at these two places I change the coding style.
So to go with minimal changes and maximum consistency I would suggest to keep it as is.

Reply:
Another alternative solution would be to just unbind the note-off midi in the XML.

Reply:
Interesting. How?

Reply:
@Swiftb0y Buttons might be used for different purposes, i.e. for layering. I don't recommend to ignore MIDI signals in the XML, which should unconditionally forward signals to JS. Except those controls that are connected directly to COs. Subsequent filtering in JS is more versatile.

Reply:
> Buttons might be used for different purposes, i.e. for layering. 

Yes, but there is only a single binding per message in the XML as far as I know, so the layering would have to be done in the function we are currently discussing, which is not the case. So while I would generally agree with your statement in more complex code, it would be ok in this case. Though I agree that it might trip up other people trying to change the mapping in the future. Consequently I agree that Holzhaus suggestion is the most ideal solution.

Reply:
Ok, as I understand this discussion, it should be fine either way (Holzhaus or mine).
So I would like to suggest to pull it as is.
Then I will mark this conversation as resolved, to proceed with the fix.
Is this ok?

Reply:
Yes. We'd still prefer if you could rebase on 2.3 though so the newer linters run. There is no benefit in merging into 2.2.

Reply:
Yes, but Holzhaus also said it shouldn't hurt to merge it from 2.2.
Since we are talking just about just one line, the linters will run.

Reply:
Checked it out locally, pre-commit checks passed.

Reply:
Thank you!